pkg/proxy/iptables/proxier.go

```diff
@@ -204,11 +204,8 @@ func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
 		encounteredError = true
 	}
 
-	args = []string{"-m", "comment", "--comment", "kubernetes service traffic requiring SNAT", "-m", "mark", "--mark", iptablesMasqueradeMark, "-j", "MASQUERADE"}
-	if err := ipt.DeleteRule(utiliptables.TableNAT, utiliptables.ChainPostrouting, args...); err != nil {
-		glog.Errorf("Error removing pure-iptables proxy rule: %v", err)
-		encounteredError = true
-	}
+	// remove all masquerade rules from the POSTROUTING chain.
+	encounteredError = ensureMasqueradeRuleOnly(ipt, "") || encounteredError
```

Comment:
I think we can still do it this way, but the transition is annoying.  Making the change to a new-chain means that we can eventually delete the transition, though.

How about this.  Unilaterally delete the old-mark rule, suppress errors, add a TODO to remove that block after 1.2 is cut.  Add a KUBE-APPLY-MASQ chain, and a unilateral jump from POSTROUTING to KUBE-APPLY-MASQ.  And appropriate cleanup.  Then on startup we can simply save and restore that chain with the new value.

Oh puke, this means all the rules that set the mark ALSO need to change on every kube-proxy restart.  Blech.

We could do a similar thing - have a KUBE-SET-MASQ chain that sets the marks, so we can save/restore and make all the KUBE-SEP chains jump to that rather than setting the mark directly.

Again, the transition is annoying, but we get to delete it.

Am I asking too much?  This is harder than I expected (I'm going to have that written on my gravestone, I think)